modules/ROOT/edits/execution-engine.adoc

```diff
@@ -0,0 +1,273 @@
+# Only applies when org.mule.runtime.scheduler.threadPool.strategy=DEDICATED
+org.mule.runtime.scheduler.io.threadPool.threadKeepAlive=30000
+
+# The number of threads to keep in the cpu_intensive pool, even if they are idle.
+# Supports Expressions
+# Only applies when org.mule.runtime.scheduler.threadPool.strategy=DEDICATED
+org.mule.runtime.scheduler.cpuIntensive.threadPool.size=2*cores
+
+# The size of the queue to use for holding cpu_intensive tasks before they are executed.
+# Supports Expressions
+# Only applies when org.mule.runtime.scheduler.threadPool.strategy=DEDICATED
+org.mule.runtime.scheduler.cpuIntensive.workQueue.size=2*cores
+----
+
+==== Considerations
+
+MuleSoft doesn’t recommend changing the used pool strategy nor change its configuration values. Note that the configuration is global and affects the entire Mule Runtime instance.
+
+MuleSoft recommends that you perform load and stress testing with all applications involved in real-life scenarios to validate any change in the threading configurations and to understand how the thread pools work in Mule 4.
+
+=== Configuration at the Application Level
+
+You can define the pool strategy to use in an application by adding the following to your application code:
+
+[source, xml, linenums]
+----
+<ee:scheduler-pools poolStrategy="UBER" gracefulShutdownTimeout="15000">
+   <ee:uber
+       corePoolSize="1"
+       maxPoolSize="9"
+       queueSize="5"
+       keepAlive="5"/>
+</ee:scheduler-pools>
+----
+
+Notice that there’s a `poolStrategy` parameter. This parameter exists for backwards compatibility reasons and allows you to switch back to the three pools scheme from prior versions of mule:
+
+[source, xml, linenums]
+----
+<ee:scheduler-pools gracefulShutdownTimeout="15000">
+   <ee:cpu-light
+           poolSize="2"
+           queueSize="1024"/>
+   <ee:io
+           corePoolSize="1"
+           maxPoolSize="2"
+           queueSize="0"
+           keepAlive="30000"/>
+   <ee:cpu-intensive
+           poolSize="4"
+           queueSize="2048"/>
+</ee:scheduler-pools>
+----
+
+==== Considerations
+
+Applying pool configurations at the application level causes Mule to create a completely new set of thread pools for the Mule app. This configuration does not change the default settings configured in the `scheduler-conf.properties` file, which is particularly important for on-premises deployments because many Mule apps can be deployed to the same Mule instance.
+
+MuleSoft recommends running Mule using default settings.
+If you are upgrading from Mule 4.1.x or Mule 4.2.x you can use the DEDICATED strategy so that any customizations applied in the `scheduler-conf.properties` file can still be used. However, MuleSoft still recommends to try the new default setting first and then determine whether the optimizations are still required. Also, try first to customize the new UBER strategy before falling back to the legacy mode.
```

Comment:
```suggestion
If you upgrade from Mule 4.1.x or Mule 4.2.x you can use the DEDICATED strategy so that customizations to `scheduler-conf.properties` are still used. However, MuleSoft recommends that you test this default setting to determine whether the optimizations are still required. Also, try first to customize the new UBER strategy before reverting to the legacy mode.
```